### PR TITLE
Writing cleaner code, improving the look and feel and adding the necessary files and libraries.

### PR DESCRIPTION
**Pull request by MaxBrillant**
In this branch, I was able to improve the overall database access approach by cleaning up repetitive and boilerplate code related to the creation of a connection with the database. 
I also improved the overall UI or look and feel of the application by using a much modern library called Flatlaf instead of sticking to Nimbus.
Finally, I regrouped in the "JAR libraries" folder all the dependencies and JAR files that the application needs in order to run properly.

### CODE REVIEW

File: src/view/UserMgt.java [hunk 1]
```diff
@@ -511,7 +520,7 @@ private String getUserLevel() {
         }
     }
     
-    private void adminWindowLinkMouseClicked(java.awt.event.MouseEvent evt) {//GEN-FIRST:event_adminWindowLinkMouseClicked
+    private void adminWindowLinkMouseClicked(java.awt.event.MouseEvent evt) throws SQLException {//GEN-FIRST:event_adminWindowLinkMouseClicked
```
Comment (r-e-d-ant) [inline src/view/UserMgt.java:523]: Why did you add `throws SQLException` , yet we are not dealing with any Sql stuffs here?!

File: src/view/RegisterUser.java [hunk 2]
```diff
@@ -10,6 +10,7 @@
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.sql.SQLException;
```
Comment (r-e-d-ant) [inline src/view/RegisterUser.java:13]: Explain this line, why did you import it?

File: src/view/UserMgt.java (hunk 1 above)
Comment (MaxBrillant) [reply src/view/UserMgt.java:523]: It's because, in simple terms, in Java, the way exceptions work is like this: sometimes you want to catch errors when the code is not executed as expected or if there is something that is wrong. That's why, usually we surround our code with "try catch" statements. Alternatively, instead of using "try catch" statements that can make your code look hard to read and less clean, you can throw an exception at the declaration of a method.
Example: 
```
public User getUser() throws UserNotFoundException {
//write some code.
}
```
However, this approach comes with a trade-off, each time you call the method that throws an exception, you will have to throw it again in the method that is making the call 

```
public void addUser() throws UserNotFoundException {     //This method will also throw the exception
User user = getUser(); //this trows a UserNotFoundException
//write some code
}
```

This approach will ensure that all the methods that are hierarchically linked to the method that throws the exception, will also throw the exception.

File: src/view/RegisterUser.java (hunk 2 above)
Comment (MaxBrillant) [reply src/view/RegisterUser.java:13]: This line is here because we have called the Login class. And in the login class, that's where we are creating our initial and only connection with the database, as I said in the title of this PR, I wrote cleaner, faster, and shorter code by removing the need to initiate or create a database connection each time we make a request to the DB. We shall create the connection to the DB once, in the Login class, and then make requests to the DB using that same connection. That's where we need to throw the SQLException in case the connection fails. Now each time we make a request, we will need to throw the SQLException so that we can catch connection errors. Each class that instantiates or calls the Login class is going to also throw the SQLException as a result of what I just stated above.

File: src/view/UserMgt.java (hunk 1 above)
Comment (r-e-d-ant) [reply src/view/UserMgt.java:523]: i see, i was familiar with `try catch` block that's why i was a bit confused.

**Comment by r-e-d-ant** [comment]
Thanks for the look improvement with flatlaf, i didn't know it before. And I could not find a way to create the connection to the database only once at the start, but thanks now i see how.
But i am not finding where you are closing the connection??! can you clarify here how it is working.

**Comment by MaxBrillant** [comment]
> Thanks for the look improvement with flatlaf, i didn't know it before. And I could not find a way to create the connection to the database only once at the start, but thanks now I see how. But i am not finding where you are closing the connection??! can you clarify here how it is working.

Good question, I never bothered about it because I presumed that the connection would be automatically closed when the application is not running anymore. Still, I might be wrong about that, but there is nothing to worry about. You can use another approach to execute code once the application is closed.

```
public static void main(String[] args) {
    Runtime.getRuntime().addShutdownHook(new Thread(new Runnable() {
        public void run() {
           //Write code to close the connection
          Home.conn.close;
        }
    }, "Shutdown-thread"));
}
```

This will ensure that the connection is closed once you close the application.

**Comment by r-e-d-ant** [comment]
Alright @MaxBrillant ,, i see the logic. Thanks

**Review by r-e-d-ant — APPROVED** [review]
Removed redundancy, improved the look.
